modules/nw-ovn-ipsec-north-south-enable.adoc

```diff
@@ -18,16 +16,99 @@ After you apply the machine config, the Machine Config Operator reboots affected
+metadata:
+  name: ipsec-config
+spec:
+  nodeSelector:
+    kubernetes.io/hostname: "<hostname>" <1>
+  desiredState:
+    interfaces:
+    - name: <interface_name> <2>
+      type: ipsec
+      libreswan:
+        left: <cluster_node> <3>
+        leftid: '%fromcert'
+        leftrsasigkey: '%cert'
+        leftcert: left_server
+        leftmodecfgclient: false
+        right: <external_host> <4>
+        rightid: '%fromcert'
+        rightrsasigkey: '%cert'
+        rightsubnet: <external_address>/32 <5>
+        ikev2: insist
+        type: transport
+----
+<1> Specifies the host name to apply the policy to. This host serves as the left side host in the IPsec configuration.
+<2> Specifies the name of the interface to create on the host.
+<3> Specifies the host name of the cluster node that terminates the IPsec tunnel on the cluster side. The name should match SAN `[Subject Alternate Name]` from your supplied PKCS#12 certificates.
+<4> Specifies the external host name, such as `host.example.com`. The name should match the SAN `[Subject Alternate Name]` from your supplied PKCS#12 certificates.
+<5> Specifies the IP address of the external host, such as `10.1.2.3.4/32`.
+
+.Example NMState IPsec tunnel configuration
+[source,yaml]
+----
+apiVersion: nmstate.io/v1
+kind: NodeNetworkConfigurationPolicy
+metadata:
+  name: ipsec-config
+spec:
+  nodeSelector:
+    kubernetes.io/hostname: "<hostname>" <1>
+  desiredState:
+    interfaces:
+    - name: <interface_name> <2>
+      type: ipsec
+      libreswan:
+        left: <cluster_node> <3>
+        leftid: '%fromcert'
+        leftmodecfgclient: false
+        leftrsasigkey: '%cert'
+        leftcert: left_server
+        right: <external_host> <4>
+        rightid: '%fromcert'
+        rightrsasigkey: '%cert'
+        rightsubnet: <external_address>/32 <5>
+        ikev2: insist
+        type: tunnel
+----
+<1> Specifies the host name to apply the policy to. This host serves as the left side host in the IPsec configuration.
+<2> Specifies the name of the interface to create on the host.
+<3> Specifies the host name of the cluster node that terminates the IPsec tunnel on the cluster side. The name should match SAN `[Subject Alternate Name]` from your supplied PKCS#12 certificates.
+<4> Specifies the external host name, such as `host.example.com`. The name should match the SAN `[Subject Alternate Name]` from your supplied PKCS#12 certificates.
+<5> Specifies the IP address of the external host, such as `10.1.2.3.4/32`.
+--
 
-. Create an IPsec configuration file named `ipsec-endpoint-config.conf`. The configuration is consumed in the next step. For more information, see link:https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/securing_networks/configuring-a-vpn-with-ipsec_securing-networks#configuring-a-vpn-with-ipsec_securing-networks[Libreswan as an IPsec VPN implementation].
+.. To configure the IPsec interface, enter the following command:
```

Comment:
It seems we didn't mention how to remove the IPsec interface.  Refer to https://docs.openshift.com/container-platform/4.14/networking/k8s_nmstate/k8s-nmstate-updating-node-network-config.html#virt-removing-interface-from-nodes_k8s_nmstate-updating-node-network-config for other kind of interface remove

 Here is an example yaml for IPSec interface
```
kind: NodeNetworkConfigurationPolicy
apiVersion: nmstate.io/v1
metadata:
  name: <PolicyName>
spec:
  nodeSelector:
    kubernetes.io/hostname: <nodeName>
  desiredState:
    interfaces:
    - name: <tunnel_name>      //Note tunnel name should be same as the one which is to be removed
      type: ipsec
      state: absent
```

Reply:
@huiran0826, is this distinct from turning IPsec itself off?

Thanks!

Reply:
It's about how to turning IPSec off using NMState, is it already covered somewhere I missed ?

Reply:
@huiran0826, there's this: `modules/nw-ovn-ipsec-disable.adoc`

But it doesn't yet cover deleting any NMState objects. Is that part of disabling IPsec on the cluster?

So there are two things here:
- (a) Disable IPsec/set Mode=Disabled
- (b) Remove/Stop/Delete a specific IPsec tunnel to an external host

I have (a). Right now, there is no (b) in the PR. Is (b) what you're asking about?

Thanks!

Reply:
Yeah, I am talking about (b) , Tear down the IPSec tunnel which was configured by NMState